include/aikido/constraint/FrameTestable.hpp

```diff
@@ -29,7 +29,13 @@ class FrameTestable : public Testable
   /// \param _state a MetaskeletonState to set the configuration of this
   ///        constraint's metaskeketon. This state's StateSpace should match
   ///        StateSpace returend by getStateSpace().
-  bool isSatisfied(const statespace::StateSpace::State* _state) const override;
+  bool isSatisfied(
+      const statespace::StateSpace::State* _state,
+      TestableOutcome* outcome = nullptr) const override;
+
+  /// Return an instance of DefaultOutcome, since this class doesn't have a
+  /// more specialized TestableOutcome derivative assigned to it.
+  std::unique_ptr<TestableOutcome> createOutcome() const override;
```

Comment:
Instead of this logic, I suggest returning the value of `_poseConstraint->createOutcome()` to preserve any useful debugging information that may contain. This should be strictly more informative than returning a `DefaultOutcome`. See my comment on `isSatisfied()` for details.

At some point in the future, we may need to wrap this outcome in a `FrameTestable` to, e.g. include some context about the joint configuration that was used to perform the check. But that can come later. 🙃